flycheck.el

```diff
@@ -6737,6 +6738,46 @@ See URL `http://asciidoctor.org'."
             line-end))
   :modes adoc-mode)
 
+(defun flycheck-awk-gawk-fix-message (err)
+  "Remove the repeated file-name/line from the error message of ERR."
+  (setf (flycheck-error-message err)
+        (replace-regexp-in-string
+         "^\\([ \t]*\\)\\(.*\n\\)\\1"
+         "\\2"
+         (replace-regexp-in-string
+          "\ngawk: [^ ]*:"
+          "\n"
+          (flycheck-error-message err))))
+  err)
+
+(defun flycheck-awk-gawk-error-filter (errors)
+  "Remove repeated file-name/line from ERRORS."
+  (seq-do #'flycheck-awk-gawk-fix-message errors)
+  errors)
+
+(flycheck-define-checker awk-gawk
+  "GNU awk's built-in --lint checker."
+  :command ("gawk"
+            ;; Avoid code execution https://github.com/w0rp/ale/pull/1411
+            "--source" "'BEGIN{exit} END{exit 1}'"
+            "-f"
+            source-original
```

Comment:
Could we change this to `source` maybe? Does the current directory matter?

Reply:
If not, we need to add `:predicate flycheck-buffer-saved-p`, since otherwise we'll get errors for the file's on-disk contents which won't match the buffer's contents

Reply:
`source` seems to work when I test :)